config/v3_0/types/config.go

```diff
@@ -24,3 +30,49 @@ var (
 		Minor: 0,
 	}
 )
+
+func (cfg Config) Validate(c path.ContextPath) (r report.Report) {
+	systemdPath := "/etc/systemd/system/"
+	unit_paths := make([]string, 0)
+	for _, unit := range cfg.Systemd.Units {
+		if unit.Contents != nil {
```

Comment:
Unfortunately, if dropin `contents` is the empty string, we write the file, but if unit `contents` is the empty string, we don't.  Ignition generally has inconsistent behavior with empty strings.  Since this is just a warning, let's alert on only the safe cases and use `!util.NilOrEmpty()` for both of these checks.

Reply:
Updated!